text/0000-stdlib-term-mouse-input-plus.md

```diff
@@ -0,0 +1,196 @@
+- Feature Name: stdlib-term-mouse-input-plus
+- Start Date: 2022-06-20
+- RFC PR: 
+- Pony Issue: 
+
+## Summary
+
+Enhance ANSI terminal support in package `term` with mouse input handling, additional ANSI escape codes to support alt buffer and cursor switching, and capture of SIGINT and SIGTSTP to handle `^C` and `^Z` as input.
+
+**Table of contents**
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Detailed design](#detailed-design)
+  - [New `ANSINotify` methods](#new-ansinotify-methods)
+  - [New `ANSI` escape codes](#new-ansi-escape-codes)
+  - [Enhance `ANSITerm`](#enhance-ansiterm)
+    - [Mouse input handling](#mouse-input-handling)
+    - [Ctrl-C and Ctrl-Z input handling](#ctrl-c-and-ctrl-z-input-handling)
+  - [ANSI terminal options](#ansi-terminal-options)
+- [How We Teach This](#how-we-teach-this)
+- [How We Test This](#how-we-test-this)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Reference Implementation](#reference-implementation)
+- [Questions](#questions)
+- [References](#references)
+
+## Motivation
+
+> Why are we doing this? 
+
+Improve ANSI terminal application capabilities when implementing text-based UIs.
+
+> What use cases does it support? 
+
+Mouse input handling, alt buffer switching, cursor hiding and capturing missing control keys.
+
+> What is the expected outcome?
+
+Enable richer text UI applications with both key and mouse input and improved user experience.
+
+## Detailed design
+
+> This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the language to understand, and for somebody familiar with the compiler to implement. This should get into specifics and corner-cases, and include examples of how the feature is used.
+
+### New `ANSINotify` methods
+
+`fun ref mouse_down(button: U8, x: U32, y: U32)`
+* This method is called whenever a mouse button is pressed
+* `button` is either 0 (left), 1 (middle), 2 (right)
+* `x` and `y` are the character coordinates below the mouse cursor
+
+`fun ref mouse_up(button: U8, x: U32, y: U32) => None` 
+* This method is called whenever a mouse button is released
+* `button` is either 0 (left), 1 (middle), 2 (right)
```

Comment:
Can we make button be enumerated primitives that use 0/1/2 underneath? See https://tutorial.ponylang.io/types/type-aliases.html#enumerations if you do not know what I am suggesting.

Reply:
@rhagenson, while this proposal addresses the three mouse buttons, in the Xterm specs it's is possible for the button number to reflect more than left, middle, right (because of advanced mice with title buttons etc). My thinking was that for now I'll limit it to the known three button numbers, but over time if we handle additional buttons then naming will become hard as the high buttons don't always mean the same thing across fancy mice. 😄  

I don't like to "predict the future" often; in this case since the API maps underlying xterm/ANSI concepts 1 for 1, I propose using numeric values.